### PR TITLE
fix(sensor_calibration_manager): fixed sensor calibration manager related issues

### PR DESCRIPTION
**Pull request by knzo25**
## Description

Upon the release of the new api, two problems were encountered:
 - When closing the sensor calibration tools or sending sigint/sigkill to the terminal, sometimes zombie processes remained
 - When the sensor calibration manager started and there were no tfs available, the state machine got stuck and never became able to calibrate
## Related links

Tickets:
 - [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-6047)
 - [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-6046)

## Tests performed

After the `Launcher configuration`  widget, the sensor calibration manager executes the respective launchers. In this condition, even if we close the sensor calibration manager via widget or console, no zombies remain

(can be checked with `ps -aux | grep ros`) or similar

Previously, if we ran the sensor calibration manager until after the launchers were executed with no autoware nor rosbag (which publishes the /tf_static), even if we started autoware or the rosbag afterwards, the calibration tool never became able to calibrate due to the design of the internal state machine. With this PR, is should be fixed

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [ x The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: sensor/sensor_calibration_manager/sensor_calibration_manager/calibrator_base.py [hunk 1]
```diff
@@ -48,32 +48,54 @@ def __init__(self, ros_interface: RosInterface):
...
+        if self.state == CalibratorState.WAITING_TFS and self.tfs_ready:
+            self.state = (
+                CalibratorState.READY
+                if self.services_available
+                else CalibratorState.WAITING_SERVICES
+            )
+
+        if prev_state != self.state:
+            self.state_changed_signal.emit(self.state)
+        pass
```
Comment (vividf) [inline sensor/sensor_calibration_manager/sensor_calibration_manager/calibrator_base.py]: pass is not needed.

File: sensor/sensor_calibration_manager/sensor_calibration_manager/sensor_calibration_manager.py [hunk 2]
```diff
@@ -292,20 +296,32 @@ def tf_graph_callback2(self, tfs_dict):
...
             )
 
+    def terminate_calibrators(self):
+        if self.process:
+            process = psutil.Process(self.process.pid)
+            for proc in process.children(recursive=True):
+                proc.kill()
+
+            process.kill()
+            logging.info("Lanch process terminated")
```
Comment (vividf) [inline sensor/sensor_calibration_manager/sensor_calibration_manager/sensor_calibration_manager.py]: Launch

**Review by vividf — CHANGES_REQUESTED** [review]
Looks good!
Only need some small fixes.

File: sensor/sensor_calibration_manager/sensor_calibration_manager/calibrator_base.py (hunk 1 above)
Comment (knzo25) [reply sensor/sensor_calibration_manager/sensor_calibration_manager/calibrator_base.py]: Thanks. Addressed in [9243c2f](https://github.com/tier4/CalibrationTools/pull/170/commits/9243c2f41b475ed821ec446a940560d463346328)

File: sensor/sensor_calibration_manager/sensor_calibration_manager/sensor_calibration_manager.py (hunk 2 above)
Comment (knzo25) [reply sensor/sensor_calibration_manager/sensor_calibration_manager/sensor_calibration_manager.py]: Thanks. Addressed in [9243c2f](https://github.com/tier4/CalibrationTools/pull/170/commits/9243c2f41b475ed821ec446a940560d463346328)

**Review by vividf — APPROVED** [review]
LGTM!

**Comment by knzo25** [comment]
@vividf 
Did you confirm that the changes fixes the issues on your environment? (the two issues this PR addresses)

**Comment by vividf** [comment]
@knzo25 
Yes, the button will be enabled even /tf comes later, also all of the processes are killed.

**Comment by knzo25** [comment]
@vividf Thanks for the check !

**Pull request merged**